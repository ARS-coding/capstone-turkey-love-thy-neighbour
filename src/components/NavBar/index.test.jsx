import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import Navbar from ".";

describe("Navbar", () => {
  const mockState = {
    user: {
      authCred: {
        user: {
          uid: "Test Uid",
        },
      },
      isSignedIn: true,
    },
  };
  const mockStore = configureStore();
  let store;
  let tree;

  beforeEach(() => {
    store = mockStore(mockState);
    tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Navbar />
          </Router>
        </Provider>
      )
      .toJSON();
  });

  it("renders Navbar correctly", () => {
    expect(tree).toMatchSnapshot();
  });
});
