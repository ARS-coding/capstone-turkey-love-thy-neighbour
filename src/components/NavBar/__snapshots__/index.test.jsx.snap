// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders Navbar correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <svg
        className="logo-svg"
      >
        logoGrayBg.svg
      </svg>
    </a>
    <div
      className="d-flex order-lg-last"
    >
      <div
        className="dropdown"
        id="notifications-dropdown"
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle btn btn-success"
          disabled={false}
          id="dropdown-basic"
          onClick={[Function]}
          type="button"
        >
          <svg>
            notification-bell.svg
          </svg>
        </button>
      </div>
    </div>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="order-lg-3 navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="me-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          href="/profile/undefined"
          onClick={[Function]}
        >
          Profile
        </a>
        <a
          className="nav-link"
          href="/meet"
          onClick={[Function]}
        >
          Meet
        </a>
        <a
          className="nav-link"
          href="/neighbors"
          onClick={[Function]}
        >
          Neighbors
        </a>
      </div>
      <div
        className="ms-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Sign Out
        </a>
      </div>
    </div>
  </div>
</nav>
`;
